pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go

```diff
@@ -1841,242 +1846,299 @@ func getNumberOfPodInterfaces(spec *v1.VirtualMachineInstanceSpec) int {
-				Field:   field.Index(idx).String(),
-			})
-		}
-
-		// Verify pci address
-		if disk.Disk != nil && disk.Disk.PciAddress != "" {
-			if disk.Disk.Bus != v1.DiskBusVirtio {
-				causes = append(causes, metav1.StatusCause{
-					Type:    metav1.CauseTypeFieldValueInvalid,
-					Message: fmt.Sprintf("disk %s - setting a PCI address is only possible with bus type virtio.", field.Child("domain", "devices", "disks", "disk").Index(idx).Child("name").String()),
-					Field:   field.Child("domain", "devices", "disks", "disk").Index(idx).Child("pciAddress").String(),
-				})
-			}
-
-			_, err := hwutil.ParsePciAddress(disk.Disk.PciAddress)
-			if err != nil {
-				causes = append(causes, metav1.StatusCause{
-					Type:    metav1.CauseTypeFieldValueInvalid,
-					Message: fmt.Sprintf("disk %s has malformed PCI address (%s).", field.Child("domain", "devices", "disks", "disk").Index(idx).Child("name").String(), disk.Disk.PciAddress),
-					Field:   field.Child("domain", "devices", "disks", "disk").Index(idx).Child("pciAddress").String(),
-				})
-			}
-		}
+func validatePciAddress(field *k8sfield.Path, idx int, disk v1.Disk) []metav1.StatusCause {
+	var causes []metav1.StatusCause
+	if disk.Disk == nil || disk.Disk.PciAddress == "" {
+		return causes
+	}
 
-		// Verify boot order is greater than 0, if provided
-		if disk.BootOrder != nil && *disk.BootOrder < 1 {
-			causes = append(causes, metav1.StatusCause{
-				Type:    metav1.CauseTypeFieldValueInvalid,
-				Message: fmt.Sprintf("%s must have a boot order > 0, if supplied", field.Index(idx).String()),
-				Field:   field.Index(idx).Child("bootOrder").String(),
-			})
-		}
+	if disk.Disk.Bus != v1.DiskBusVirtio {
+		causes = append(causes, metav1.StatusCause{
+			Type:    metav1.CauseTypeFieldValueInvalid,
+			Message: fmt.Sprintf("disk %s - setting a PCI address is only possible with bus type virtio.", field.Child("domain", "devices", "disks", "disk").Index(idx).Child("name").String()),
+			Field:   field.Child("domain", "devices", "disks", "disk").Index(idx).Child("pciAddress").String(),
+		})
+	}
 
-		// Verify bus is supported, if provided
-		if len(bus) > 0 {
-			if bus == "ide" {
-				causes = append(causes, metav1.StatusCause{
-					Type:    metav1.CauseTypeFieldValueInvalid,
-					Message: "IDE bus is not supported",
-					Field:   field.Index(idx).Child(diskType, "bus").String(),
-				})
-			} else {
-				buses := []v1.DiskBus{v1.DiskBusVirtio, v1.DiskBusSCSI, v1.DiskBusSATA, v1.DiskBusUSB}
-				validBus := false
-				for _, b := range buses {
-					if b == bus {
-						validBus = true
-					}
-				}
-				if !validBus {
-					causes = append(causes, metav1.StatusCause{
-						Type:    metav1.CauseTypeFieldValueInvalid,
-						Message: fmt.Sprintf("%s is set with an unrecognized bus %s, must be one of: %v", field.Index(idx).String(), bus, buses),
-						Field:   field.Index(idx).Child(diskType, "bus").String(),
-					})
-				}
+	_, err := hwutil.ParsePciAddress(disk.Disk.PciAddress)
```

Comment:
Could be inlined into following if-condition.

Reply:
Have made the requested changes